michael-rob/starter-code/scripts/articleView.js

```diff
@@ -0,0 +1,155 @@
+        scrollTop: ($(this).parent().offset().top)
+      },200);
+      $(this).html('Read on &rarr;');
+      $(this).parent().find('.article-body *:nth-of-type(n+2)').hide();
+    }
+  });
+};
+
+articleView.initNewArticlePage = function() {
+  // DONE: Ensure the main .tab-content area is revealed. We might add more tabs later or otherwise edit the tab navigation.
+  $('.tab-content').show();
+
+  // DONE: The new articles we create will be copy/pasted into our source data file.
+  // We need to allow the container to generate the JSON that will be copy-pasted.
+  // We can hide it for now, and show it once we have data to export.
+  $('#article-export').hide();
+
+  $('#article-json').on('focus', function(){
+    this.select();
+  });
+
+  // DONE: Add an event handler to...
+  //  - update the preview field if any inputs change -- DONE
+  //  - update the export field if any inputs change.
+  $('form').on('change', function(event){
+    /*
+    - Grab the template's HTML
+    - Compile the template into a callable function
+    - Call the function with the data with which you want to fill the template
+    - Append the filled template to the DOM
+    */
+    var sourceHTML = $('#preview-template').html();
+    var compiledTemplate = Handlebars.compile(sourceHTML);
+
+    // if the published box is checked, then calculate how many days ago this article was published, and put that in my article preview.
+    // if the published box is not checked, then just put the text "(draft)" in my article preview
+
+    var today = new Date();
+    var dateText = today.toLocaleDateString(); // this is for the JSON text later. Figure out where to put it
+
+    var formData = {
+      title: this.title.value,
+      body: this.body.value,
+      author: this.author.value,
+      url: this.url.value,
+      publishStatus: this.publishStatus.checked ? `published 0 days ago` : '(draft)',
+      category: this.category.value
+    };
+    var fillTemplate = compiledTemplate(formData);
+    $('#articles')
+      .empty()
+      .append(fillTemplate);
+    // Stringify our form data
+    // Grab the disabled input field
+    // Fill the field with our JSON string
+    $('#article-json').val(JSON.stringify(formData));
+    $('#article-export').show();
+  })
+
+};
+
+articleView.create = function() {
+  // DONE: Set up a var to hold the new article we are creating.
+  // Clear out the #articles element, so we can put in the updated preview
+  var newArticle;
+  $('#articles').html('');
+
+  // DONE: Instantiate an article based on what's in the form fields:
+  newArticle = new Article (JSON.parse($('#article-json')[0].value));
+
+  // DONE: Use our interface to the Handblebars template to put this new article into the DOM:
+  $('#articles').append(newArticle.toHtml());
+
+  // DONE: Activate the highlighting of any code blocks; look at the documentation for hljs to see how to do this by placing a callback function in the .each():
+  $('pre code').each();
```

Comment:
The callback function for the highlighting is missing. Check the handlebars documentation.